### ISSUE TITLE
[TF2] Some multicore commands don't work

### ISSUE THREAD

**Issue by BreavyTF2** [ValveSoftware/Source-1-Games#4641]
The commands 
cl_threaded_client_leaf_system
r_threaded_client_shadow_manager
r_threaded_renderables
are all not used, the code has their uses disabled with if(0) or commented out.
These commands do function on the modified team-comtress-2 repository, and are reenabled by default.

`		bool bThreaded = false;//( nDirty > 5 && cl_threaded_client_leaf_system.GetBool() && g_pThreadPool->NumThreads() );`
`	m_bThreaded = false;//( r_threaded_client_shadow_manager.GetBool() && g_pThreadPool->NumIdleThreads() );`
`	if ( 0 && r_threaded_renderables.GetBool() )
	{
		ParallelProcess( "BoneSetupNpcsLast", arrBoneSetupNpcsLast.Base() + numOpaqueEnts - numNpcs, numNpcs, &SetupBonesOnBaseAnimating );
		ParallelProcess( "BoneSetupNpcsLast NonNPCs", arrBoneSetupNpcsLast.Base(), numNonNpcsAnimating, &SetupBonesOnBaseAnimating );
	}`

**Comment by BreavyTF2** [comment]
> It'd be great to have more multi-threaded code in tf2, although team comtress 2 is built using illegally obtained tf2 source code so i personally wouldn't recommend mentioning it here or posting leaked tf2 source code here especially when asking valve devs about

yes, but people have posted direct lines and fixes using the leaked code, the code wasn't illegally obtained, it was illegally shared, the code was provided to developers with a source 1 sdk license.

**Comment by BreavyTF2** [comment]
also boosting the thread count from 4 to 6 would be good as well.

**Comment by RicochetYT** [comment]
It would be nice to see any optimizations that don't break stuff, ported or implemented into TF2 or any other Source games. I think the best way of futureproofing TF2 is to take advantage of all the resources hardware has to provide, but Source was never great about that, and I'm not even sure what I'm talking about!

**Comment by MikhailaTF** [comment]
I'd love for a the optimizations, QoL-changes, bugfixes and general effort that went into big projects such as team comtress to eventually and carefully be implemented into TF2!

**Comment by condor00fr** [comment]
> I'd love for a the optimizations, QoL-changes, bugfixes and general effort that went into big projects such as team comtress to eventually and carefully be implemented into TF2!

We all do honestly. What Mastercom and her team managed to create is one of the biggest tf2 project ever made, the amout of optimisation and bug fixes in this is enormous and Mastercom, who is already popular with the Masterconfig, is a source magician;)

**Comment by mastercoms** [comment]
I don't recommend using those commands, they're just enabled for some testing on my end. Most of them cause pretty bad bottlenecks from ordering issues and incorrect task balancing. Just because multi-threading is theoretically faster doesn't mean it will be.
